### ISSUE TITLE
Hydrant Status: missing fields

### ISSUE THREAD

**Issue by rssidlowski** [JMurk/Valve_Hydrant_Mobile_Issues#47]
Missing the following fields:
Missing/Damage Cap?
CSR ID
SR ID
Work Order ID

**Comment by yousifmalik** [comment]
Please provide more details about these missing fields. Where to place them? which fields in db they map too?

**Comment by yousifmalik** [comment]
![hydrant 1](https://cloud.githubusercontent.com/assets/13753096/13608247/efb3c7a6-e521-11e5-9d11-a24415d3f123.png)
![hydrant 2](https://cloud.githubusercontent.com/assets/13753096/13608248/efb3ef10-e521-11e5-9fe2-4d2faf12f6f5.png)

**Comment by rssidlowski** [comment]
It should be 'Missing/Damaged Cap?' and place after 'Missing/Damaged Chains?'.
The other fields should be after Pitot (PSI).

Scott, can you see if these fields are in the database?

**Comment by yousifmalik** [comment]
Currently, I have the following fields mapping:
CRS ID -> CSR_SR_NUM
SR ID -> CW_SR_NUM
Work Order ID -> WO_NUM
Please verify if this mapping

**Comment by JMurk** [comment]
And Missing/Damaged Cap -> MISSING_CAP_YN

The others you noted are correct.

**Comment by slmessier** [comment]
Yousif - **Please also update the _SR ID_ & _Work Order ID_ attribute values to function as hyperlinks** using the specifications described below:

**UX Example** (note the green box):
![image](https://cloud.githubusercontent.com/assets/4098544/13608689/cda1382c-e523-11e5-9403-85ab5e711fcb.png)

**SR URL Text & Example** (UX passes ID value into URL string):
https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=%2fCityworks%2fWorkManagement%2fRequestEdit.aspx%3fRequestId%3d218108&RequestId=218108
![image](https://cloud.githubusercontent.com/assets/4098544/13608773/4540fbba-e524-11e5-8cb7-ca1c82db5d6e.png)

**Work Order URL Text & Example** (UX passes ID value into URL string):
https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=%2fCityworks%2fWorkManagement%2fWoGeneralEdit.aspx%3fWorkOrderId%3d293183&WorkOrderId=293183
![image](https://cloud.githubusercontent.com/assets/4098544/13608846/7b78c55a-e524-11e5-9260-a732351df3d5.png)

**Comment by yousifmalik** [comment]
Ok, How do we want to display CRS ID, SR ID, and Work Order ID? are they suppose to be on their own section (Associated Work Activities)? 
Is "CSR ID" field read-only?

**Comment by slmessier** [comment]
Yes these new attributes should be presented within a distinct container / section within the UX as visualized within the graphic / mockup above.

CSR ID is not read-only. it should remain editable, however the hyperlink functionality does not need to be applied to that item.

**Comment by yousifmalik** [comment]
Ready for testing on DEV

**Comment by slmessier** [comment]
Thanks Yousif - I can confirm that the hyperlink functionality is working however the hyperlink is not linking to the correct URL's. The hyperlink is currently erroneously linking to the Valve/Hydrant URL.

Please update the target URL for the SR ID and Work Order ID hyperlinks to utilize the correct URLs appropriately. The URL's are provided above within this issue, note each has a distinct URL.

I am using Hydrant # 152C010003 to test. you can reference the 2 most recent inspections for that hydrant to validate.

**Comment by yousifmalik** [comment]
I am setting the following URLs where the {id} will be replace by the actual ID value:
For "SR ID",
https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=%2fCityworks%2fWorkManagement%2fRequestEdit.aspx%3fRequestId%3d{id}&RequestId={id}  

For "Work Order",
https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=%2fCityworks%2fWorkManagement%2fWoGeneralEdit.aspx%3fWorkOrderId%3d{id}&WorkOrderId={id}

Are these the correct URLs and parameters? 
I am seeing "RequestId" and "WorkOrderId" URL parameters are duplicated on "SR ID" and "Work Order" URLs respectively.

**Comment by rssidlowski** [comment]
They are correct URLs.  Note the second RequestId does not have the '%3d'.

**Comment by yousifmalik** [comment]
ok, i will change '=' with '%3d' on both URLs. Please confirm these are good URLs:

https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=%2fCityworks%2fWorkManagement%2fRequestEdit.aspx%3fRequestId%3d218108&RequestId%3d218108

https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=%2fCityworks%2fWorkManagement%2fWoGeneralEdit.aspx%3fWorkOrderId%3d293183&WorkOrderId%3d293183

**Comment by JMurk** [comment]
Those are correct, but could you update the URL once more from /Cityworks/ to /CityworksTest/ ?

**Comment by yousifmalik** [comment]
I am seeing two instances of "Cityworks". Can you you provide the exact url using these templates:
https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=/Cityworks/WorkManagement/RequestEdit.aspx?RequestId={id}&RequestId={id}

https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=/Cityworks/WorkManagement/WoGeneralEdit.aspx?WorkOrderId={id}&WorkOrderId={id}

**Comment by yousifmalik** [comment]
Links for SR and work order no longer working on DEV. the code working ok on my local system and when i push to dev the links are not set!

Any changes i should be aware of on dev? 

I have these debugging lines:
Local DEV system:

targetUrl: 
https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=/Cityworks/WorkManagement/RequestEdit.aspx?RequestId={id}&RequestId={id}

Generated href: 
https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=/Cityworks/WorkManagement/RequestEdit.aspx?RequestId=106C000559&RequestId=106C000559

Update html: <a href="https://dpwapps.baltimorecity.gov/Cityworks/Login.aspx?ReturnUrl=/Cityworks/WorkManagement/RequestEdit.aspx?RequestId=106C000559&RequestId=106C000559" target="_blank">106C000559</a>

DEV Server shown empty url:
targetUrl: 
Generated href: 
Update html: <a href="" target="_blank">106C000559</a>

**Comment by JMurk** [comment]
First- the test URLs should be:

SRs:  https://dpwapps.baltimorecity.gov/CityworksTest/WorkManagement/RequestEdit.aspx?RequestId=[id]

WOS: https://dpwapps.baltimorecity.gov/CityworksTest/WorkManagement/WoGeneralEdit.aspx?WorkOrderId=[id]

I'm not sure what you mean by they are not set or not working?  I can't even edit the fields in the UX for testing...

**Comment by yousifmalik** [comment]
Links for SR and WO used to navigate ok to whatever url they were configure with and they are not now!

**Comment by JMurk** [comment]
I still am not understanding.  You can call me if you'd like - 410-316-7800 x1505

**Comment by yousifmalik** [comment]
OK, problem solved! it was due to the url encoding during the app build for dev environment.

**Comment by yousifmalik** [comment]
Completed and ready for testing on DEV

**Comment by slmessier** [comment]
I was able to successfully validate the functionality in the development instance. This issue will be updated to "Ready for Production".
